src/wallet/wallet.ts

```diff
@@ -581,7 +592,9 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     // We should wait for new addresses before setting wallet to ready
     await this.getNewAddresses(true);
 
-    this.setupConnection();
+    if (this.isWsEnabled()) {
```

Comment:
Do we really need `this.isWsEnabled`?

I mean, the idea of using the `DummyConnection` is to be able to call the methods from it with no effect, like this `this.setupConnection`